data/moves.js

```diff
@@ -9917,15 +9914,14 @@ let BattleMovedex = {
 				}
 				if (move && (move.target === 'self' || move.category === 'Status')) return;
 				this.add('-activate', target, 'move: Mat Block', move.name);
-				source.moveThisTurnResult = true;
 				let lockedmove = source.getVolatile('lockedmove');
 				if (lockedmove) {
 					// Outrage counter is reset
 					if (source.volatiles['lockedmove'].duration === 2) {
 						delete source.volatiles['lockedmove'];
 					}
 				}
-				return null;
+				return 0;
```

Comment:
ok, `0` is clearly a sentinel value here, and it's not one I particularly like.

I think we might actually want some sort of enum here. `Battle.PREVENT_MOVE_NO_FAIL`?